haskell/src/Lentille/GitHub/PullRequests.hs

```diff
@@ -0,0 +1,553 @@
+                    )
+                      { changeEventAuthor = Just $ getIdent actor,
+                        changeEventCreatedAt = Just $ from createdAt
+                      }
+               in Just event
+          SearchNodesTimelineItemsNodesHeadRefForcePushedEvent
+            _
+            _
+            createdAt
+            (Just (SearchNodesTimelineItemsNodesFpactorActor _ actor)) ->
+              Just
+                ( baseEvent
+                    (ChangeEventTypeChangeCommitForcePushed ChangeCommitForcePushedEvent)
+                    ("ChangeCommitForcePushedEvent-" <> changeId change)
+                    change
+                )
+                  { changeEventAuthor = Just $ getIdent actor,
+                    changeEventCreatedAt = Just $ from createdAt
+                  }
+          _ -> Nothing
+    getCommitEvents :: Change -> SearchNodesCommitsPullRequestCommitConnection -> [ChangeEvent]
+    getCommitEvents change (SearchNodesCommitsPullRequestCommitConnection _ nodes) =
+      toEvent <$> catMaybes (fromMaybe [] nodes)
+      where
+        toEvent :: SearchNodesCommitsNodesPullRequestCommit -> ChangeEvent
+        toEvent (SearchNodesCommitsNodesPullRequestCommit commit) =
+          ( baseEvent
+              (ChangeEventTypeChangeCommitPushed ChangeCommitPushedEvent)
+              ("ChangeCommitPushedEvent-" <> changeId change)
+              change
+          )
+            { changeEventAuthor = Just $ getCommitter $ committer commit,
+              changeEventCreatedAt = maybe (changeCreatedAt change) (Just . from) (pushedDate commit)
+            }
+          where
+            getCommitter :: Maybe SearchNodesCommitsNodesCommitCommitterGitActor -> Ident
+            getCommitter
+              ( Just
+                  ( SearchNodesCommitsNodesCommitCommitterGitActor
+                      (Just (SearchNodesCommitsNodesCommitCommitterUserUser login))
+                    )
+                ) = getIdent login
+            getCommitter _ = getGhostIdent
+    getCommentEvents :: Change -> SearchNodesCommentsIssueCommentConnection -> [ChangeEvent]
+    getCommentEvents change (SearchNodesCommentsIssueCommentConnection nodes) =
+      catMaybes $ toEvent <$> catMaybes (fromMaybe [] nodes)
+      where
+        toEvent :: SearchNodesCommentsNodesIssueComment -> Maybe ChangeEvent
+        toEvent
+          ( SearchNodesCommentsNodesIssueComment
+              (ID eId)
+              createdAt
+              (Just (SearchNodesCommentsNodesAuthorActor _ actor))
+            ) =
+            Just
+              ( baseEvent
+                  (ChangeEventTypeChangeCommented ChangeCommentedEvent)
+                  ("ChangeCommmentedEvent-" <> changeId change <> "-" <> from eId)
+                  change
+              )
+                { changeEventAuthor = Just $ getIdent actor,
+                  changeEventCreatedAt = Just $ from createdAt
+                }
+        toEvent _ = Nothing
```

Comment:
And what about this case?

Reply:
For this one removing the line raises a warning about an incomplete pattern match. However the first and unique match is the one we need. So the default match seems to be needed to satisfied the type checker.